### PR TITLE
chore(aws-eks): update integration test stacks to be env agnostic

### PR DESCRIPTION
**Pull request by corymhall**
----

### All Submissions:

* [ ] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/master/INTEGRATION_TESTS.md)?
	* [ ] Did you use `cdk-integ` to deploy the infrastructure and generate the snapshot (i.e. `cdk-integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-eks/test/integ.eks-cluster-handlers-vpc.ts [hunk 1]
```diff
@@ -1,12 +1,12 @@
 /// !cdk-integ pragma:ignore-assets
-import { App } from '@aws-cdk/core';
+import { App, Stack } from '@aws-cdk/core';
 import * as eks from '../lib';
-import { TestStack } from './util';
+// import { TestStack } from './util';
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-eks/test/integ.eks-cluster-handlers-vpc.ts:4]: leftover?

File: packages/@aws-cdk/aws-eks/test/integ.eks-helm-asset.ts [hunk 2]
```diff
@@ -19,7 +18,7 @@ class EksClusterStack extends TestStack {
     });
 
     // just need one nat gateway to simplify the test
-    this.vpc = new ec2.Vpc(this, 'Vpc', { maxAzs: 3, natGateways: 1 });
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-eks/test/integ.eks-helm-asset.ts:22]: why remove `maxAzs`?

File: packages/@aws-cdk/aws-eks/test/integ.eks-inference.ts [hunk 3]
```diff
@@ -1,16 +1,15 @@
...
 
-class EksClusterInferenceStack extends TestStack {
+class EksClusterInferenceStack extends Stack {
 
   constructor(scope: App, id: string) {
     super(scope, id);
 
     // just need one nat gateway to simplify the test
-    const vpc = new ec2.Vpc(this, 'Vpc', { maxAzs: 3, natGateways: 1 });
+    const vpc = new ec2.Vpc(this, 'Vpc', { maxAzs: 2, natGateways: 1 });
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-eks/test/integ.eks-inference.ts:12]: why change this from 3 to 2?

File: packages/@aws-cdk/aws-eks/test/integ.alb-controller.ts [hunk 4]
```diff
@@ -1,19 +1,18 @@
...
 
-class EksClusterAlbControllerStack extends TestStack {
+class EksClusterAlbControllerStack extends Stack {
 
   constructor(scope: App, id: string) {
     super(scope, id);
 
     // just need one nat gateway to simplify the test
-    const vpc = new ec2.Vpc(this, 'Vpc', { maxAzs: 3, natGateways: 1 });
+    const vpc = new ec2.Vpc(this, 'Vpc', { maxAzs: 2, natGateways: 1 });
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-eks/test/integ.alb-controller.ts:15]: why change this from 3 to 2?

**Review by comcalvi — CHANGES_REQUESTED** [review]
looks great! Just one minor comment.

File: packages/@aws-cdk/aws-eks/test/integ.eks-helm-asset.ts (hunk 2 above)
Comment (corymhall) [reply packages/@aws-cdk/aws-eks/test/integ.eks-helm-asset.ts:22]: for environment agnostic stacks `maxAzs` can only be `2`.

**Review by comcalvi — APPROVED** [review]
Thanks Cory, nice work!

**Pull request merged**